compiler/qsc_eval/src/intrinsic.rs

```diff
@@ -36,6 +37,15 @@ pub(crate) fn call(
         #[allow(clippy::cast_precision_loss)]
         "IntAsDouble" => Ok(Value::Double(arg.unwrap_int() as f64)),
         "IntAsBigInt" => Ok(Value::BigInt(BigInt::from(arg.unwrap_int()))),
+        "DoubleAsStringWithPrecision" => {
+            let [input, prec_val] = unwrap_tuple(arg);
+            let precision = usize::try_from(prec_val.unwrap_int()).expect("integer value");
+            Ok(Value::String(Rc::from(format!(
+                "{:.*}",
+                precision,
+                input.unwrap_double()
+            ))))
```

Comment:
This may require a little extra handling for round numbers as `input` as well as the `precision` zero case. Since the only thing that differentiates the display of an integer and round numbered double in Q# is the presence of a decimal point, you'll want to make sure that no matter values are passed in the given number always has at least one decimal place. For example, `DoubleAsStringWithPrecision(1.0, 2)` needs to return the string `"1.00"` and `DoubleAsStringWithPrecision(1.1, 0)` should either return an error or always use a single zero decimal, so return `"1.0"`.

Reply:
It could be, though that would be a new convention for Q#, there isn't anything else that displays a double without something after the `.`.

Reply:
As-is looks like `DoubleAsStringWithPrecision(1.1, 0)` returns "1". Would it be better to leave it as Rust implements it or return "1.0"?

Reply:
I think it at least needs a decimal point to differentiate it from an integer. So either "1." or "1.0" would be preferred.